pkg/filter/filter.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cuelang.org/go/cue"
+	"github.com/kubevela/pkg/cue/cuex"
+)
+
+// ApplyFilter applies the given filter to an object.
+func ApplyFilter(ctx context.Context, contextData map[string]interface{}, filter string) (bool, error) {
+	contextByte, err := json.Marshal(contextData)
+	if err != nil {
+		return false, err
+	}
+	contextTemplate := fmt.Sprintf("context: %s", string(contextByte))
+	compiler := cuex.DefaultCompiler.Get()
+	filterVal, err := compiler.CompileString(ctx, fmt.Sprintf("filter: {\n%s\n%s\n}", filter, contextTemplate))
+	if err != nil {
+		return false, err
+	}
+	if filterVal.Err() != nil {
+		return false, filterVal.Err()
+	}
+	result := filterVal.LookupPath(cue.ParsePath("filter"))
+	if filterVal.LookupPath(cue.ParsePath("filter.filter")).Exists() {
```

Comment:
why filter.filter? for compatibility? P.S. I would suggest make some const value for "filter" and "filter.filter".

Reply:
The first the filter is hidden for the user, I'll wrap the cue value into `filter` to evaluate its result. Users can also write a complicated logic in `filter` and set the result in filter like:

```
filter: |
   if a != _|_ && a.b == "b" {
      filter: true
    }
    if a == _|_ {
      filter: false
    }
```

Reply:
ok